### ISSUE TITLE
Image properties are managed by themes

### ISSUE THREAD

**Issue by ericmorand** [NightlyCommit/magento2-module-soc#1]
# Violation

Magento 2 famework forces image properties to be managed by themes.

# Why it is a violation of the Separation of Concerns principles

Image properties are declared in `etc/view.xml`, a file that is expected by Magento 2 framework to be provided by themes. This is documented in [Configure theme properties](https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/theme-images.html).

This is a violation because image properties depend on _image types_, and _image types_ are product attributes created either by modules or from the back-office. It makes the themes dependent on data that they are not managing themselves and can't be declared as dependencies.

# Illustration of the violation

Consider the following `etc/view.xml` file, taken from the above documentation:

```xml
<images module="Magento_Catalog">
    <image id="unique_image_id" type="image">
        <width>100</width> <!-- Image width in px -->
        <height>100</height> <!-- Image height in px -->
    </image>
</images>
```

Here, a theme declares an _image identifier_ named `unique_image_id` as a resized variant of the `image` _image type_. This `image` _image type_ may or may not exist: if one of the installed module did create that `image` product attribute, or if that product attribute was created in the back-office, then `unique_image_id` is a perfectly valid _image identifier_; if no module did create the `image` product attribute and it was not created either from the back-office, then `unique_image_id` is not valid and trying to use it to create an image variant would lead to a crash.

It is impossible when writing a theme to know if a product attribute exists - and thus it is impossible to know if an _image identifier_ declared in a theme will function or not.

In other words, having _image identifiers_ managed by themes makes the behavior of the final product unpredictable.

# Workarounds

It may be possible to declare image properties programmatically at modules level. This needs to be checked. If not, there is no workaround for this issue.

# Proposed solution

* Override Magento 2 configuration resolvers to have image properties declared by modules instead of themes.
* Remove support for `etc/view.xml` by themes configuration resolvers.
